src/serialization/json_object.cpp

```diff
@@ -146,6 +146,26 @@ void fromJsonValue(const rapidjson::Value& val, std::string& str)
   str = val.GetString();
 }
 
+void toJsonValue(rapidjson::Writer<epee::byte_stream>& dest, const std::vector<std::uint8_t>& src)
+{
+  const std::string hex = epee::to_hex::string(epee::to_span(src));
+  dest.String(hex.data(), hex.size());
+}
+
+void fromJsonValue(const rapidjson::Value& val, std::vector<std::uint8_t>& dest)
+{
+  if (!val.IsString())
+  {
+    throw WRONG_TYPE("string");
+  }
+
+  dest.resize(val.GetStringLength() / 2);
+  if (!epee::from_hex::to_buffer(epee::to_mut_span(dest), {val.GetString(), val.GetStringLength()}))
+  {
+    throw BAD_INPUT();
+  }
+}
```

Comment:
If you send "f" as val, this resizes dest to 0, and to_buffer seems to do the same thing, it looks like you'll get an empty result.

Reply:
`from_hex::to_buffer` checks the destination buffer size before writing, and returns false on size mismatches. I'll add another check here for readability + maintenance churn proofing.

Reply:
Ah you're right, it's only to_string which doesn't check for odd.